### PR TITLE
doc: nrf: add section comparing MQTT and REST

### PR DESCRIPTION
**Pull request by jayteemo**
Add section to the nRF Cloud documentation that describes and
compares MQTT and REST so that a user can choose the correct
protocol for their project.
CIA-518

Signed-off-by: Justin Morton <justin.morton@nordicsemi.no>

### CODE REVIEW

**Review by plskeggs — APPROVED** [review]

File: doc/nrf/ug_nrf_cloud.rst [hunk 1]
```diff
@@ -24,6 +24,33 @@ You can use the services offered by nRF Cloud in the following scenarios:
...
 #. Device connected to a customer cloud service in a suitable manner. The services can be used from the customer cloud service that communicates over REST to the nRF Cloud REST API in a proxy configuration.
 
+Choosing a protocol: MQTT or REST
+*********************************
+
+REST Overview:
+* The device initiates a TLS connection to nRF Cloud.
+* nRF Cloud supports a connection keep-alive/idle time of 60 seconds for REST API sockets.
+* For authentication, the device must send a JSON Web Token (JWT) with each REST transaction.  The JWT is approximately 450 bytes, but can be larger depending on the claims.
+* Each REST transaction contains HTTP headers, including the JWT, and any API specific payload.
```
Comment (peknis) [inline doc/nrf/ug_nrf_cloud.rst:34]:
```suggestion
REST overview
=============

* The device initiates a TLS connection to nRF Cloud.
* nRF Cloud supports a connection keep-alive/idle time of 60 seconds for REST API sockets.
* For authentication, the device must send a JSON Web Token (JWT) with each REST transaction. 
  The JWT is approximately 450 bytes, but can be larger depending on the claims.
* Each REST transaction contains HTTP headers, including the JWT, and any API specific payload.
```

File: doc/nrf/ug_nrf_cloud.rst [hunk 2]
```diff
@@ -24,6 +24,33 @@ You can use the services offered by nRF Cloud in the following scenarios:
...
+
+MQTT Overview:
+* The device initiates a mutual-TLS (mTLS) connection to the nRF Cloud MQTT broker.
+* The MQTT keep-alive time can be set by the device and can be longer than 60s.
+* Device authentication via mTLS lasts throughout the MQTT connection.
+* Once connected, the device subscribes to the desired MQTT topics.
+* Each MQTT publish event contains the MQTT topic and the payload.
```
Comment (peknis) [inline doc/nrf/ug_nrf_cloud.rst:41]:
```suggestion
MQTT overview
=============

* The device initiates a mutual-TLS (mTLS) connection to the nRF Cloud MQTT broker.
* The MQTT keep-alive time can be set by the device and can be longer than 60s.
* Device authentication through mTLS lasts throughout the MQTT connection.
* Once connected, the device subscribes to the desired MQTT topics.
* Each MQTT publish event contains the MQTT topic and the payload.
```

File: doc/nrf/ug_nrf_cloud.rst [hunk 3]
```diff
@@ -24,6 +24,33 @@ You can use the services offered by nRF Cloud in the following scenarios:
...
+
+MQTT has a higher (data/power) cost to setup a connection.  However, the data size per MQTT publish event is smaller than a comparable REST transaction.
+MQTT may be preferred if a device is able to maintain a connection to the broker and sends/receives data frequently.
+REST may be preferred if a device sends data infrequently or does not need to receive unsolicited data from the cloud.
```
Comment (peknis) [inline doc/nrf/ug_nrf_cloud.rst:45]: How about having this information right after the considerations below?

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/ug_nrf_cloud.rst [hunk 4]
```diff
@@ -24,6 +24,33 @@ You can use the services offered by nRF Cloud in the following scenarios:
...
+* Each REST transaction contains HTTP headers, including the JWT, and any API specific payload.
+
+MQTT Overview:
+* The device initiates a mutual-TLS (mTLS) connection to the nRF Cloud MQTT broker.
+* The MQTT keep-alive time can be set by the device and can be longer than 60s.
+* Device authentication via mTLS lasts throughout the MQTT connection.
+* Once connected, the device subscribes to the desired MQTT topics.
+* Each MQTT publish event contains the MQTT topic and the payload.
+
+MQTT has a higher (data/power) cost to setup a connection.  However, the data size per MQTT publish event is smaller than a comparable REST transaction.
```
Comment (peknis) [inline doc/nrf/ug_nrf_cloud.rst:43]:
```suggestion
MQTT has a higher (data/power) cost to set up a connection.  However, the data size of a MQTT publish event is smaller than a comparable REST transaction.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by jtguggedal — APPROVED** [review]
This is very useful, thanks!

**Review by peknis — APPROVED** [review]

**Pull request merged**